### PR TITLE
The update dialog is not shown after interrupting downloading

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Please have a look at our [guidelines for contributions](https://github.com/microsoft/appcenter-sdk-android/blob/develop/CONTRIBUTING.md) and consider the following before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [x] Did you add unit tests?
* [x]  Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?
* [x] Did you check UI tests on the sample app? They are not executed on CI.

## Description

A few sentences describing the overall goals of the pull request.

1. Open application
2. The update dialog is shown. Click downloading.
3. During downloading disable distribute a module. The downloading is stopped.
4. Enable distribute a module. The update dialog is not shown.
5. Collapse and expand the app.
6. The update dialog still is not shown.

Expected
After 4 steps the updates dialog is shown.

Actual
After 4 steps the updates dialog is not shown.

### CODE REVIEW

File: sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeTest.java [hunk 1]
```diff
@@ -671,6 +675,36 @@ public void showMandatoryDownloadReadyDialogTest() {
...
+        verify(mHttpClient).callAsync(anyString(), anyString(), eq(Collections.<String, String>emptyMap()), any(HttpClient.CallTemplate.class), httpCallback.capture());
+
+        /* Complete call with no new release (this will return the default mock mReleaseDetails with version 0). */
+        httpCallback.getValue().onCallSucceeded(mock(HttpResponse.class));
+
+        /* Disable and enable distribute module. */
+        Distribute.setEnabled(false);
+        Distribute.setEnabled(true);
+
+        /* Verify download is not called again. */
```
Comment (MatkovIvan) [inline sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeTest.java:704]: > is not called again

`times(2)` ? Update comment or check. BTW did this test fail without the fix?

Comment (AnastasiaKubova) [reply sdk/appcenter-distribute/src/test/java/com/microsoft/appcenter/distribute/DistributeTest.java:704]: yes, it fail without the fix

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,6 +15,7 @@ ___
 * **[Feature]** Add a `disableAutomaticCheckForUpdate` API that needs to be called before SDK start in order to turn off automatic check for update. 
 * **[Feature]** Add a `checkForUpdate` API to manually check for update.
 * **[Fix]** Fix not checking updates on public track if it was ignored on error while initializing in-app update for private track.
+* **[Fix]** Fix checking updates after disabling the Distribute module during downloading the release.
```
Comment (guperrot) [inline CHANGELOG.md:18]:
```suggestion
* **[Fix]** Fix checking for updates after disabling the Distribute module while downloading the release.
```

Comment (guperrot) [reply CHANGELOG.md:18]: Wrong version section.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,7 @@
 ### App Center Crashes
 
 * **[Fix]** Remove the multiple attachments warning as that is now supported by the portal.
+* **[Fix]** Fix checking for updates after disabling the Distribute module while downloading the release.
```
Comment (MatkovIvan) [inline CHANGELOG.md:8]:
```suggestion

### App Center Distribute

* **[Fix]** Fix checking for updates after disabling the Distribute module while downloading the release.
```

**Pull request merged**